### PR TITLE
Add Code Numbered component

### LINKED ISSUE #36: Build 'Code numbered' component in React
[Vanilla docs reference](https://docs.vanillaframework.io/en/patterns/code-numbered)

### LINKED ISSUE #37: Add 'Code numbered' story to Storybook
[Vanilla docs reference](https://docs.vanillaframework.io/en/patterns/code-numbered)

### LINKED ISSUE #38: Add testing to 'Code numbered' component
[Vanilla docs reference](https://docs.vanillaframework.io/en/patterns/code-numbered)

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
# Done
- Added CodeBlock component with optional "numbered" modifier
- Added story to Storybook
- Added snapshot tests

# QA
- Pull code
- Run `yarn clean`, `yarn build` and `yarn serve`
- Run `yarn lint` and ensure there are no linting errors
- Verify the CodeBlock and CodeBlock Numbered pattern in Storybook
- Run `yarn test` and ensure the CodeBlock tests pass

# Fixes
Fixes #36, fixes #37, fixes #38

### CODE REVIEW

File: src/lib/components/CodeNumbered/CodeNumbered.test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ReactTestRenderer from 'react-test-renderer';
+import CodeNumbered from './CodeNumbered';
+
+describe('CodeNumbered component', () => {
+  it('should compare with a snapshot', () => {
+    const codenumbered = ReactTestRenderer.create(<CodeNumbered lines={[
```
Comment (tbille) [inline src/lib/components/CodeNumbered/CodeNumbered.test.js]: `camelCase` var names

**Review by barrymcgee — COMMENTED** [review]
I'm not sure passing the code snippet as an array of lines as a named prop is the best approach given this could be used to display some very large code blocks. 

An alternative approach would be to add each line as children of the component, wrapped in `<code>` snippets and iterate over the `this.props.children` object instead?

**Review by Lukewh — APPROVED** [review]
👍 love it. Thanks @Caleb-Ellis

**Comment by Lukewh** [comment]
@Caleb-Ellis @barrymcgee's point is valid, though I'd maybe go a step further and accept unformatted text and split on new lines and create the array from that?

**Comment by Caleb-Ellis** [comment]
@barrymcgee good point, didn't really consider how large the code blocks might get. I'll give @Lukewh's idea a go

**Review by Lukewh — CHANGES_REQUESTED** [review]
Looks good but could we add a test for a 1 liner?

**Comment by tbille** [comment]
* Not sure if it is really relevant but what happens if I use `tabs` instead of `spaces` for indentation?
If this is relevant it would need a test.
* And what happens if it don't pass code in the component? Is it displayed with no lines? -> I think you need a test for this use case.

**Comment by Lukewh** [comment]
Also what happens if someone does pass in some markup - we should escape it.

**Comment by Caleb-Ellis** [comment]
I've changed the component a bit by making it a more generic Code Block with an optional modifier of 'numbered'. Thoughts?
@Lukewh You should be able to pass markup as children now, which will be converted to a string and prettified automatically.
@tbille If nothing is passed through an code block with no lines will show. There are now tests for this as well as single line + multiline codeblocks, with/without the numbered modifier

**Comment by tbille** [comment]
@Caleb-Ellis `If nothing is passed through, a code block with no lines will show. ` Is it the behavior you want? Wouldn't it be better to have one empty line?

Add lines between tests to make your file more readable :smiley:

**Comment by Caleb-Ellis** [comment]
@tbille Oops, that's a typo - one empty line will show

**Review by Lukewh — APPROVED** [review]
Looks good thanks @Caleb-Ellis 👍

**Pull request merged**